Packs/SentryWire/Integrations/SentryWire/SentryWire.py

```diff
@@ -0,0 +1,327 @@
+        node_str += f"{NodeName},"
+    node_str = node_str[:-1]
+
+    readable_output = f"SearchID : {SearchID}\nNodeName(s) : {node_str}"
+
+    outputs["SearchID"] = SearchID
+
+    if generate_links:
+        readable_output += "\n"
+        outputs["checkstatus"] = []
+        outputs["getpcaps"] = []
+        outputs["metadata"] = []
+        for i in range(len(response)):
+            for key in response[i]:
+                value = str(response[i].get(key))
+                readable_output += f"{key} : {value}\n"
+            readable_output = readable_output[:-1]
+            outputs["checkstatus"].append(response[i].get("checkstatus"))
+            outputs["getpcaps"].append(response[i].get("getpcaps"))
+            outputs["metadata"].append(response[i].get("metadata"))
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='SentryWire.Investigator.Search',
+        outputs_key_field='SearchID',
+        outputs=outputs
+    )
+
+
+def delete_search_command(client: Client, args: Dict[str, Any]):
+    SearchID = args.get("SearchID")
+    response = client.delete_search(search_id=str(SearchID))
+    response["SearchID"] = SearchID
+    return CommandResults(
+        readable_output=f"{SearchID} has been deleted!",
+        outputs_prefix='SentryWire.Investigator.Deleted',
+        outputs_key_field='SearchID',
+        outputs=response
+    )
+
+
+def download_pcap_command(client: Client, args: Dict[str, Any]):
+    SearchID = args.get('SearchID')
+    nodename = args.get('NodeName')
+    file_entry = fileResult(
+        filename=f'{SearchID}.pcap',
+        data=client.download_pcap(search_id=str(SearchID), node_name=str(nodename)).content
+    )
+    return_results(file_entry)
+
+
+def download_metadata_command(client: Client, args: Dict[str, Any]):
+    SearchID = args.get('SearchID')
+    nodename = args.get('NodeName')
+    file_entry = fileResult(
+        filename=f'{SearchID}.zip',
+        data=client.download_metadata(search_id=str(SearchID), node_name=str(nodename)).content
+    )
+    return_results(file_entry)
+
+
+def get_search_status_command(client: Client, args: Dict[str, Any]):
+    SearchID = args.get('SearchID')
+    nodename = args.get('NodeName')
+
+    # Status request
+    response = client.get_search_status(search_id=str(SearchID), node_name=str(nodename))
+
+    # Add SearchID/NodeName
+    response["SearchID"] = SearchID
+
+    # Remove redundant/superfluous data
+    redundant = ["SearchName",
+                 "SearchKey",
+                 "ID",
+                 "CaseName",
+                 "MasterToken",
+                 "SearchPorts",
+                 "SubmittedTime",
+                 "MaxChunk",
+                 "SearchType"]
+    for key in redundant:
+        try:
+            response.pop(key)
+        except KeyError:
+            pass
+
+    # Parse status
+    # Very ugly, will need to fix
+    if "SearchResult" in response:

```

Comment:
Maybe outsource the logic to a helper function, if you need any help please update me, I'll be happy to help.